src/symtab2gb/symtab2gb_parse_options.cpp

```diff
@@ -0,0 +1,138 @@
+/******************************************************************\
+
+Module: symtab2gb_parse_options
+
+Author: Diffblue Ltd.
+
+\******************************************************************/
+
+#include "symtab2gb_parse_options.h"
+
+#include <fstream>
+#include <iostream>
+#include <string>
+
+#include <goto-programs/goto_convert_functions.h>
+#include <goto-programs/goto_model.h>
+#include <goto-programs/link_goto_model.h>
+#include <goto-programs/write_goto_binary.h>
+#include <json-symtab-language/json_symtab_language.h>
+#include <util/exception_utils.h>
+#include <util/exit_codes.h>
+#include <util/invariant.h>
+#include <util/optional.h>
+#include <util/version.h>
+
+symtab2gb_parse_optionst::symtab2gb_parse_optionst(int argc, const char *argv[])
+  : parse_options_baset{SYMTAB2GB_OPTIONS,
+                        argc,
+                        argv,
+                        std::string("SYMTAB2GB ") + CBMC_VERSION}
+{
+}
+
+static inline bool failed(bool error_indicator)
+{
+  return error_indicator;
+}
+
+static void run_symtab2gb(
+  const std::vector<std::string> &symtab_filenames,
+  const std::string &gb_filename)
+{
+  // try opening all the files first to make sure we can
+  // even read/write what we want
+  std::ofstream out_file{gb_filename};
```

Comment:
Regarding your windows CI issue, does windows still care about the difference between binary and ascii file modes, and if so, what does std::ofstream do by default? I'm wondering if you need to put the stream into a binary mode?

Reply:
src/goto-cc/compiler.cpp does this: `std::ofstream outfile(file_name, std::ios::binary);` so I suspect you might need the binary mode.

Reply:
Also, do you need to explicitly close() the output file as well?

Reply:
`binary` yes, I had not considered that at all but now that you mention it that does seem to be the kind of thing that'd cause this problem.

`close` I don't think so, the `std` streams are all RAII controlled, so they should close even on exception.

Reply:
Could you actually add a precondition to `write_goto_binary` that tests that the stream is open in binary mode?

Reply:
@tautschnig AFAIK iostreams don't expose a way to query or change the openmode